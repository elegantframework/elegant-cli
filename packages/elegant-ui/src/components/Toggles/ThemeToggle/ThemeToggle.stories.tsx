import type { Meta, StoryObj } from '@storybook/react';
import ThemeToggle from './ThemeToggle';


const meta: Meta<typeof ThemeToggle> = {
  title: 'Core/Toggles/Theme Toggle',
  component: ThemeToggle,
};

export default meta;

type Story = StoryObj<typeof ThemeToggle>;

export const Default: Story = {
  args: {
    panelClassName: "mt-4"
  },
};