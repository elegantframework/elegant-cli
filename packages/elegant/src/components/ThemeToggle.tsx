"use client"

import { useState, useEffect } from "react"
import { useTheme } from "next-themes"
import { Moon, Sun, Laptop } from "lucide-react"

import { Button } from "@/components/ui/button"
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu"
import { cn } from "@/utils/utils"

export default function ThemeSelector() {
  const [mounted, setMounted] = useState(false)
  const { theme, setTheme } = useTheme()

  useEffect(() => {
    setMounted(true)
  }, [])

  if (!mounted) {
    return null
  }

  return (
    <DropdownMenu>
      <DropdownMenuTrigger asChild>
        <Button variant="ghost" size="icon" className="w-9 px-0">
          <Sun className="h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0 fill-primary-400/20 stroke-primary-500" />
          <Moon className="absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100 fill-primary-400/20 stroke-primary-500" />
          <span className="sr-only">Toggle theme</span>
        </Button>
      </DropdownMenuTrigger>
      <DropdownMenuContent align="end">
        <DropdownMenuItem onClick={() => setTheme("light")}>
          <Sun className={cn(
            "mr-2 h-4 w-4",
            theme === "light" ? "ill-primary-400/20 stroke-primary-500" : ""
          )} />
          <span className={cn(theme === "light" ? "text-primary-500" : "")}>Light</span>
        </DropdownMenuItem>
        <DropdownMenuItem onClick={() => setTheme("dark")}>
          <Moon className={cn(
            "mr-2 h-4 w-4",
            theme === "dark" ? "ill-primary-400/20 stroke-primary-500" : ""
          )} />
          <span className={cn(theme === "dark" ? "text-primary-500" : "")}>Dark</span>
        </DropdownMenuItem>
        <DropdownMenuItem onClick={() => setTheme("system")}>
          <Laptop className={cn(
            "mr-2 h-4 w-4",
            theme === "system" ? "ill-primary-400/20 stroke-primary-500" : ""
          )} />
          <span className={cn(theme === "system" ? "text-primary-500" : "")}>System</span>
        </DropdownMenuItem>
      </DropdownMenuContent>
    </DropdownMenu>
  )
}