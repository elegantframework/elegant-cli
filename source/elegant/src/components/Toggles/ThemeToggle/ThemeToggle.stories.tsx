import type { Meta, StoryObj } from '@storybook/react';
import ThemeToggle from '@/components/Toggles/ThemeToggle/ThemeToggle';


const meta: Meta<typeof ThemeToggle> = {
  title: 'Core/Toggles/Theme Toggle',
  component: ThemeToggle,
};

export default meta;

type Story = StoryObj<typeof ThemeToggle>;

export const Default: Story = {
  args: {
    panelClassName: "mt-4"
  },
};