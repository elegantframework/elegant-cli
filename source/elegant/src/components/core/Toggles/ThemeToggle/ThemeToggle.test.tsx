import React from "react";
import renderer from 'react-test-renderer';
import ThemeToggle from "./ThemeToggle";

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
});

describe("Theme toggle component", () => {
    it('renders a theme toggle component properly', () => {
        const toggle = renderer
        .create(<ThemeToggle />)
        .toJSON();
        expect(toggle).toMatchSnapshot();
    });
});