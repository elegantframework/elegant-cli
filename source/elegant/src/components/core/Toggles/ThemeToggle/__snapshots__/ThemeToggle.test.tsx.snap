// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme toggle component renders a theme toggle component properly 1`] = `
[
  <label
    className="sr-only"
    data-headlessui-state=""
    id="headlessui-listbox-label-:r0:"
    onClick={[Function]}
  >
    Theme
  </label>,
  <button
    aria-expanded={false}
    aria-haspopup={true}
    data-headlessui-state=""
    disabled={false}
    id="headlessui-listbox-button-:r1:"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    type="button"
  >
    <span
      className="dark:hidden"
    >
      <svg
        className="w-6 h-6"
        fill="none"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
      >
        <path
          className="stroke-slate-400 dark:stroke-slate-500"
          d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
        />
        <path
          className="stroke-slate-400 dark:stroke-slate-500"
          d="M12 4v1M17.66 6.344l-.828.828M20.005 12.004h-1M17.66 17.664l-.828-.828M12 20.01V19M6.34 17.664l.835-.836M3.995 12.004h1.01M6 6l.835.836"
        />
      </svg>
    </span>
    <span
      className="hidden dark:inline"
    >
      <svg
        className="w-6 h-6"
        fill="none"
        viewBox="0 0 24 24"
      >
        <path
          className="fill-transparent"
          clipRule="evenodd"
          d="M17.715 15.15A6.5 6.5 0 0 1 9 6.035C6.106 6.922 4 9.645 4 12.867c0 3.94 3.153 7.136 7.042 7.136 3.101 0 5.734-2.032 6.673-4.853Z"
          fillRule="evenodd"
        />
        <path
          className="fill-slate-400 dark:fill-slate-500"
          d="m17.715 15.15.95.316a1 1 0 0 0-1.445-1.185l.495.869ZM9 6.035l.846.534a1 1 0 0 0-1.14-1.49L9 6.035Zm8.221 8.246a5.47 5.47 0 0 1-2.72.718v2a7.47 7.47 0 0 0 3.71-.98l-.99-1.738Zm-2.72.718A5.5 5.5 0 0 1 9 9.5H7a7.5 7.5 0 0 0 7.5 7.5v-2ZM9 9.5c0-1.079.31-2.082.845-2.93L8.153 5.5A7.47 7.47 0 0 0 7 9.5h2Zm-4 3.368C5 10.089 6.815 7.75 9.292 6.99L8.706 5.08C5.397 6.094 3 9.201 3 12.867h2Zm6.042 6.136C7.718 19.003 5 16.268 5 12.867H3c0 4.48 3.588 8.136 8.042 8.136v-2Zm5.725-4.17c-.81 2.433-3.074 4.17-5.725 4.17v2c3.552 0 6.553-2.327 7.622-5.537l-1.897-.632Z"
        />
        <path
          className="fill-slate-400 dark:fill-slate-500"
          clipRule="evenodd"
          d="M17 3a1 1 0 0 1 1 1 2 2 0 0 0 2 2 1 1 0 1 1 0 2 2 2 0 0 0-2 2 1 1 0 1 1-2 0 2 2 0 0 0-2-2 1 1 0 1 1 0-2 2 2 0 0 0 2-2 1 1 0 0 1 1-1Z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </button>,
]
`;
